src/index.ts

```diff
@@ -52,17 +52,38 @@ export const temporal = (<TState>(
 
     const curriedHandleSet =
       options?.handleSet?.(
-        (store.temporal.getState() as _TemporalState<TState>)._handleSet,
+        (store.temporal.getState() as _TemporalState<TState>)
+          ._handleSet as StoreApi<TState>['setState'],
```

Comment:
Why is this typecase needed? Is that because this function technically takes four parameters including `currentState` and `deltaState`?

Reply:
Yeah that is right. I believe it is required due to the fact that `curriedHandleSet` is used to call either the internal handle set or the user-provided handleSet.

Perhaps this is a circular explanation, but options.handleSet takes an argument of type `StoreApi<TState>['setState']`.
However, this is not the type of `_handleSet`, which takes four arguments. So asserting it to the be the type expected by options.handleSet resolves the type error.

I'll be the first to admit that this part of the code is not the easiest to read/follow. I don't immediately have an alternative suggestion, but I would certainly be open to something more approachable in upcoming versions.